src/contract/abi.h

```diff
@@ -2,263 +2,306 @@
-      std::string encodeAddressArr(const std::vector<Address>& addV) const;
+  /**
+   * Encode a boolean array into Solidity ABI format.
+   * @param bV The boolean array to encode.
+   * @return The encoded bool[] hex string, with the proper offsets and lengths.
+   */
+  std::string encodeBoolArr(const std::vector<bool> &bV) const;
 
-      /**
-       * Encode a boolean array into Solidity ABI format.
-       * @param bV The boolean array to encode.
-       * @return The encoded bool[] hex string, with the proper offsets and lengths.
-       */
-      std::string encodeBoolArr(const std::vector<bool>& bV) const;
+  /**
+   * Encode a raw bytes or UTF-8 string array into Solidity ABI format.
+   * See `encodeBytes()` for more details.
+   * @param bytesV The raw bytes or UTF-8 string array to encode.
+   * @return The encoded bytes[] or string[] hex string, with the proper offsets
+   * and lengths.
+   */
+  std::string encodeBytesArr(const std::vector<std::string> &bytesV) const;
 
-      /**
-       * Encode a raw bytes or UTF-8 string array into Solidity ABI format.
-       * See `encodeBytes()` for more details.
-       * @param bytesV The raw bytes or UTF-8 string array to encode.
-       * @return The encoded bytes[] or string[] hex string, with the proper offsets and lengths.
-       */
-      std::string encodeBytesArr(const std::vector<std::string>& bytesV) const;
+public:
+  /// Alias for variant type, for easier handling.
+  typedef std::vector<std::variant<
+      uint256_t, std::vector<uint256_t>, Address, std::vector<Address>, bool,
+      std::vector<bool>, std::string, std::vector<std::string>>>
+      EncVar;
 
-    public:
-      /// Alias for variant type, for easier handling.
-      typedef std::vector<std::variant<
-        uint256_t, std::vector<uint256_t>, Address, std::vector<Address>,
-        bool, std::vector<bool>, std::string, std::vector<std::string>
-      >> EncVar;
+  /**
+   * Constructor.
+   * Automatically encodes the data during construction.
+   * Throws on error.
+   * @param data A list of variables to encode.
+   * @param func (optional) The full function header to encode.
+   *             Defaults to an empty string.
+   *@throws std::runtime_error if the function header is invalid or if header
+   *and data do not match.
+   */
+  Encoder(const ABI::Encoder::EncVar &data, std::string_view func = "");
 
-      /**
-       * Constructor.
-       * Automatically encodes the data during construction.
-       * Throws on error.
-       * @param data A list of variables to encode.
-       * @param func (optional) The full function header to encode.
-       *             Defaults to an empty string.
-       *@throws std::runtime_error if the function header is invalid or if header
-       *and data do not match.
-       */
-      Encoder(const ABI::Encoder::EncVar& data, std::string_view func = "");
+  /// Getter for `data`.
+  /// TODO: rename this to getData() or something similar (for consistency with
```

Comment:
Ita's probably working on this in another branch? If yes please ignore this comment.